File: src/js/memoryBenchmark.js
var MemoryBenchmark = /** @class */ (function (_super) {
	__extends(MemoryBenchmark, _super);

	function MemoryBenchmark(writer, printToConsole) {
		printToConsole = printToConsole || true;
		var _this = _super.call(this, writer, printToConsole) || this;
		return _this;
	}

	MemoryBenchmark.prototype.BenchImplementation = function () {
		return this.BenchRandomMemory();
	};

	MemoryBenchmark.prototype.PopulateResult = function (benchResult, result) {
		benchResult.Points = result.Average * this.Ratio;
		benchResult.Result = result.Average;
		benchResult.Units = "MB/s";
		benchResult.Output = result.Output;
		return benchResult;
	};

	return MemoryBenchmark;
}(MemoryBenchmarkBase));

if (typeof module !== 'undefined' && module.exports) {
	module.exports = {
		MemoryBenchmark: MemoryBenchmark
	};
}

File: src/lua/generic/memoryBenchmark.lua
MemoryBenchmark = class(MemoryBenchmarkBase, function(a, writer, printToConsole)
    MemoryBenchmarkBase.init(a, writer, printToConsole)
    a.name = "MemoryBenchmark"
end)

function MemoryBenchmark:benchImplementation()
    return self:benchMemory()
end

function MemoryBenchmark:populateResult(benchResult, memoryBenchmarkResult)
    benchResult.points = memoryBenchmarkResult.average * self.ratio
    benchResult.result = memoryBenchmarkResult.average
    benchResult.units = "MB/s"
    benchResult.output = memoryBenchmarkResult.output
    return benchResult;
end